test/test_t2star.py

```diff
@@ -0,0 +1,250 @@
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Test T2Star experiment
+"""
+import unittest
+import numpy as np
+
+from qiskit.utils import apply_prefix
+from qiskit.providers import BaseBackend
+from qiskit.providers.models import QasmBackendConfiguration
+from qiskit.result import Result
+from qiskit.test import QiskitTestCase
+from qiskit_experiments.composite import ParallelExperiment
+from qiskit_experiments.characterization import T2StarExperiment
+
+
+# Fix seed for simulations
+SEED = 9000
+
+# from Yael
+class T2Backend(BaseBackend):
+    """
+    A simple and primitive backend, to be run by the T1 tests
+    """
+
+    def __init__(
+        self, p0=None, initial_prob_plus=None, readout0to1=None, readout1to0=None, dt_factor=1
+    ):
+        """
+        Initialize the T2star backend
+        """
+
+        configuration = QasmBackendConfiguration(
+            backend_name="t2star_simulator",
+            backend_version="0",
+            n_qubits=int(1e6),
+            basis_gates=["barrier", "h", "p", "delay", "measure"],
+            gates=[],
+            local=True,
+            simulator=True,
+            conditional=False,
+            open_pulse=False,
+            memory=False,
+            max_shots=int(1e6),
+            coupling_map=None,
+            dt=dt_factor,
+        )
+
+        self._t2star = p0["t2star"]
+        self._a_guess = p0["a_guess"]
+        self._f_guess = p0["f_guess"]
+        self._phi_guess = p0["phi_guess"]
+        self._b_guess = p0["b_guess"]
+        self._initial_prob_plus = initial_prob_plus
+        self._readout0to1 = readout0to1
+        self._readout1to0 = readout1to0
+        self._dt_factor = dt_factor
+        super().__init__(configuration)
+
+    # pylint: disable = arguments-differ
+    def run(self, qobj):
+        """
+        Run the T2star backend
+        """
+        shots = qobj.config.shots
+        result = {
+            "backend_name": "T2star backend",
+            "backend_version": "0",
+            "qobj_id": 0,
+            "job_id": 0,
+            "success": True,
+            "results": [],
+        }
+        for circ in qobj.experiments:
+            nqubits = circ.config.n_qubits
+            counts = dict()
+
+            if self._readout0to1 is None:
+                ro01 = np.zeros(nqubits)
+            else:
+                ro01 = self._readout0to1
+
+            if self._readout1to0 is None:
+                ro10 = np.zeros(nqubits)
+            else:
+                ro10 = self._readout1to0
+
```

Comment:
The previous lines can be moved outside of the `for` loop. Also in the T1 test.

Reply:
I don't see how this can be done - because of the dependence on `nqubits`.